README.md
# art(IF)icial Intelligence

[About](#What-is-this?)

[Content](#content)

[For advanced user](#For-advanced-user)

[Check sum](#Check-sum)

[FAQ](#FAQ)


## What is this? 

> This is avoiding objects (Obstacles) simulator/game written in java

Hey, this is how auto-car works (just kidding, if this is how, maybe there are many car crashes [but you get the point])

## known issues

>Sometimes, it does not go as expected

~~for ob > 1 obstacles: sometimes the object just does not go as expected due to the algorithm~~

~~for ob = 3 obstacles: sometimes the object just does not go as expected due to the algorithm~~

## BETA

_Discontinued_

Beta version is developed [here](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/src/master/)

## TODO: 

~~[top  priority] Change the way update works by using a string embedded instead of a separate file (almost done)~~

fix, improve algorithm, add 2 objects + 

_________________________________________

## Content

### Jar file:

[Latest here](/JarFile/AOW.jar)

### ChangeLog:

[Click here](/JarFile/ChangeLog.md)

### Pictures:

[Here](Pic/Readme.md)

______________________________________________________________

#### _ABOUT_:

**Avoiding obstacle eclipse project [here](https://github.com/frychicken/AvoidingObstacles)**

**Updater eclipse project [here](https://github.com/frychicken/UpdaterAOW)**

**Customizer eclipse project [here](https://github.com/frychicken/AOWCustomizer)**

**Debug mode:**

Display x,y,censors and everything you need to fix bugs

**Non - debug mode**

Just display moving object and obstacle(s)

**Lab(s)**

Run the program how many times you desire (default: 10; max: non; min: non)

**Obstacle(s)**

Decide how many obstacles you want (default: 1; max: 3; min: 1; default color: green)

**Coloring**

RGB color-red, green, blue: you can set this for obstacle(s) and object (default obstacle(s) color: green - R:0 G:255 B:0; default object color: red -R:255  G:0  B:0)

**sound** 

option to turn sound on or off

**auto-pilot**

You can control the object using arrow keys

press "s" to stop moving up and "c" to continue moving up

**Updates**

program will check for update automatically

**log file**

it will notify you the location; normally, it is located at your current directory

**Darkmode**

Yep, it is here

**Game mode**

This program can be a game too

**custom algorithm**

If you don't know anything, don't click check!!! (for more info, read more below)

## For advanced user

_**This is an experimental feature**_

you can change the algorithm

_Needed class to compile:_ **Soundfx, Frychicken, Estima**

Soundfx: to make sound

Frychicken: to write log 

Estima: to calculate time executed

> it is not necessary

**This is based on 2 jar commands**

``` javac``` and ``` jar uf ```

I know this is not the most efficient way

### How?

1. Download customizer (from AOW or [here](https://github.com/frychicken/art-IF-icial-Intelligence/blob/master/Customizer/AOWCus.jar?raw=true))

2. Run; first, select your .java file (you may have to write it using the template below); second, select your AOW.jar

3. IF you use Soundfx.java and/or Frychicken.java, and if you know what you are doing, type in the command to compile

4. Profit (maybe not)

#### Template for Execut.java

```java

//Logic stuff -- execut
import javax.swing.*; 
import java.awt.*;
import java.awt.event.KeyAdapter;
import java.awt.event.KeyEvent;
import java.awt.event.WindowAdapter;
import java.awt.event.WindowEvent;
import java.io.IOException;
import java.util.*;
public class Execut  extends Component {
	private static final long serialVersionUID = 1L;
	private JFrame frame;
	private Method_T contentt;
	String msg = "UP";
	private int x = 200;
	private int y = 600;
	Color object = Color.RED;
	Color obstacle = Color.GREEN;
	private boolean debugg = true;
	private boolean sound = true;
	private ArrayList<Integer> xOb = new ArrayList<Integer>();
	private  ArrayList<Integer> yOb = new ArrayList<Integer>();
	int userin = 1;
	int lalala;
	int alala;
	boolean autopilot = false;
	boolean dark;
	boolean game;
	boolean ca;
	Frychicken fry = new Frychicken();
	Estima est = new Estima();
	public Execut(int userin, Color object, Color obstacle, int i, int b, boolean darkmode, boolean ca) {
		this.object = object;
		this.obstacle = obstacle;
		lalala = i;
		alala = b;
		this.userin = userin;
		this.ca = ca;
		dark = darkmode;
	}
	public void assin(boolean debugg, boolean sound, boolean autopilot, boolean gamem) throws IOException {
		this.debugg = debugg;
		this.sound = sound;
		this.autopilot = autopilot;
		game = gamem;
		for (int i =0; i < userin; i++) {
			if (i ==0 || i ==2) {
				xOb.add(RanX());
			}else {
				double rand = Math.random();
				if (rand >= 0.5)
					xOb.add(xOb.get(0) + 237);
				else xOb.add(xOb.get(0)-237);
			}
			if (i==0||i==1)
				yOb.add(RanY());
			else {
				double rand = Math.random();
				if (rand >= 0.5)
					yOb.add(yOb.get(0) + 237);
				else yOb.add(yOb.get(0) - 237);
			}
		}
		Collections.sort(yOb);
		Collections.reverse(yOb);
		if (debugg) {
			for (int i= 0; i< userin; i++){
				System.out.println("x" + (i+1) + " " +xOb.get(i));
				fry.writeLog("x" + (i+1) + " " +xOb.get(i));
				System.out.println("y" + (i+1) +" " +yOb.get(i));
				fry.writeLog("y" + (i+1) +" " +yOb.get(i));

			}
		}
	}

	public int RanX() {
		Random randomness = new Random();
		int rand = randomness.nextInt(300)+30;
		return rand;
	}
	public int RanY() {
		Random randomness = new Random();
		int rand = randomness.nextInt(400)+30;
		return rand;
	}
	int owenn;
	int owenn2;
	class Method_T extends Component {
		private static final long serialVersionUID = 1L;
		int ccc = 0;
		public void paint(Graphics g){

			for (int i =0; i < userin; i++) {
				g.setColor(obstacle);
				g.fillRect(xOb.get(i),yOb.get(i),150,150);
				if ((y >= yOb.get(i) && y<= yOb.get(i) +150) && (x+123 >= xOb.get(i)  && x+123 <= xOb.get(i)+150) && debugg) {
					g.setColor(Color.gray);
					g.fillRect(xOb.get(i),yOb.get(i),150,150);
			owenn = y;
					owenn2 = x;
					if (ccc == 0) {
						System.out.println("Hit at x of object: "+owenn2);
						System.out.println("Hit at y of object: "+owenn);
						fry.writeLog("Hit at x of object: "+owenn2);
						fry.writeLog("Hit at y of object: "+owenn);
						ccc++;
					}
				}
				else if ((y >= yOb.get(i)  && y<= yOb.get(i) +150) && (x >= xOb.get(i)  && x <= xOb.get(i)+150) && debugg) {
					g.setColor(Color.gray);
					g.fillRect(xOb.get(i),yOb.get(i),150,150);
				owenn = y;
					owenn2 = x;
					if (ccc ==0) {
						System.out.println("Hit at x of object: "+owenn2);
						System.out.println("Hit at y of object: "+owenn);
						fry.writeLog("Hit at x of object: "+owenn2);
						fry.writeLog("Hit at y of object: "+owenn);
						ccc++;
					}
				}
			}
			if (debugg) {
				g.setColor(Color.black);
				g.fillOval(x, y, 10, 10);
				g.setColor(Color.black);
				g.fillOval(x+123, y, 10, 10);
				g.setColor(Color.red);
				g.drawString(msg, 300, 750);
				int c = 700;
				int cc = 700;

				for (int i =0; i<userin; i++) {
					g.drawString("Hit at y of object: "+owenn, 250, 660);	
					g.drawString("Hit at x of object :"+owenn2, 250, 675);
					g.drawString("y object: "+y, 150, 660);	
					g.drawString("x object: "+x, 150, 675);
					g.drawString("x"+(i+1)+": "+xOb.get(i), 500, c+=20);
					g.drawString("y"+(i+1)+": "+yOb.get(i), 430, cc+=20);
					g.drawString("lab(s) ran: "+lalala, 60, 720);
					g.drawString("Obstacle(s): "+userin, 60, 700);
					g.drawString("lab(s): "+alala, 60, 680);	
					g.drawString("auto-pilot: "+autopilot, 250, 690);	
					g.drawString("Game mode: "+game, 250, 705);	
					g.drawString("Custom Algorithm: "+ca, 250, 720);	


				}
			}
			g.setColor(object);
			g.fillRect(x + 40, y +40, 50, 50);
		}
	}
	public void run() {
		frame = new JFrame("AO Simulator" + " @Debug mode: "+debugg + ", sound: "+ sound); 
		frame.setDefaultCloseOperation(JFrame.HIDE_ON_CLOSE); 
		frame.addWindowListener(new WindowAdapter() {
			   public void windowClosing(WindowEvent evt) {
				     System.out.println();
			     System.out.println("Program terminated by user");
			     fry.writeLog("Program terminated by user");
			     est.getEst();
			     System.exit(0);
			   }
			  });
		contentt = new Method_T();
		if (dark) {
			frame.getContentPane().setBackground(Color.DARK_GRAY);  
		}
		frame.getContentPane().add(BorderLayout.CENTER, contentt);  
		frame.setIconImage(Toolkit.getDefaultToolkit().getImage(getClass().getResource("lollol.png")));
		frame.setResizable(false);
		frame.setSize(600, 800); 
		if (autopilot) {
			frame.addKeyListener(new KeyAdapter() {
				public void keyPressed(KeyEvent event) {
					moveTheTHing(event);
				}
			});
		}
		frame.setFocusable(true); 
		frame.setLocationRelativeTo(null);
		frame.setVisible(true); 
		   int firstexe = 0;
			if (game&&firstexe==0) {
		   	 up = false;
		   	 firstexe++;
		    }
		runpls();
	}


	private boolean right = false;
	private boolean left = false;
	private boolean up = true;
	private boolean down = false;
	private boolean check = true;
 
	
	public void moveTheTHing(KeyEvent event) { 
         
	
		int key = event.getKeyCode(); 
		int key2 = event.getKeyChar();
		if (key == KeyEvent.VK_LEFT) {
			x-=10;
			check = false;
			msg = "LEFT";
		}
		else if (key == KeyEvent.VK_RIGHT) {
			x+=10;
			check = false;
			msg = "RIGHT";
		}
		else if (key == KeyEvent.VK_UP) {
			y-=20;
			check = false;
			msg = "UP";
		}
		else if (key == KeyEvent.VK_DOWN) {
			y+=20;
			check = false;
			msg = "DOWN";
		}
		if (key2 == 's') {
			up = false;
			msg = "Stop";
		}
		if (key2 == 'c') {
			up = true;
			msg = "Moving";
		}

	}

	public void gameover(){
	fry.writeLog("Game over");
	System.out.println("Game over");
	est.getEst();
    try {
		fry.YouLose();
	} catch (Exception e) {
		fry.writeLog(e.toString());
	}
	frame.setVisible(false);

	Nani nina = new Nani(dark);
	try {
		nina.Nihaoma();
	} catch (Exception e) {
        fry.writeLog(e.toString());
		e.printStackTrace();
	}
	System.exit(0);
}
// Make changes to runpls() -- recommended 
	public void runpls(){
		while(true) {

			for (int i =0; i < userin; i ++) {
	if(game) {
					
					if ((y+40 >= yOb.get(i) && y+40<= yOb.get(i) +150) && (x+40 >= xOb.get(i)  && x+40 <= xOb.get(i)+150) ) {

						gameover();
					}
					else if ((y+90 >= yOb.get(i)  && y+90<= yOb.get(i) +150) && (x+90 >= xOb.get(i)  && x+90 <= xOb.get(i)+150) ) {
						gameover();
						
					}
					
				}

				if () {

					left = true;
					right = false;
				}
				else if () {
					left = false;
					right = true;
				}
				if (i==3){
					if (){
						down= true;
					}
				}
				int fff = 0;
				if(down){

				}


				if (left) {
					msg = "LEFT";
					x-=3;
					if (  ) { up = true; }
					else up = false;
					if (sound) {
						Soundfx soundfx = new Soundfx();
						soundfx.sound();
					}
				}
				if(right) {
					msg = "RIGHT";
					x+=3;
					if (  ) { up = true; }
					else up = false;
					if (sound) {
						Soundfx soundfx = new Soundfx();
						soundfx.sound();
					}
				}
				if(up) {
					y--;
					left = false;
					right = false;
				}
				if (y+40+50 <= 0) {
					if (sound) {
						Soundfx soundfx = new Soundfx();
						soundfx.sound();
					}
					frame.setVisible(false); 
					return;
				}
				try {
					Thread.sleep(3); 
				}
				catch (Exception nothing) {
					nothing.printStackTrace();
					fry.writeLog(nothing.toString());

				} 
				frame.repaint(); 
				right = false;
				left = false;
			}
		}
	}
}


```

## Check sum

[Click here](/JarFile/ChangeLog.md)

## FAQ

Where is my log file?

> It is in the .JAR file; it shouldn't be that way because of the mistake I made when making it into JAR

Why are there always errors when running the newest version?

> Because I am too lazy to test

Why it my program not updating?

> Because of the code I use -- it is not very efficient



JarFile/ChangeLog.md
## Version 1.0.0: Continue (Beta 1)

_OCT 19, 2018_

> AvoidingObstacle

Initial release

Sources:

[Obstacle.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/aa0794bb2acd93e58151d600f1a2bef1cf6e76c5/Obstacle.java)

[Execut.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/2cddfd5641fea7e95dedff6cb64394084096cee4/Execut.java)

**Sha256**: ```89c65e8876779dd5c683b80f241bf9cb57ae309de9dfd30ef167c94af7b510e7```

**MD5**: ```1958ca970568a2de086a934b73936287```

**SHA1**: ```2aa70c6bb56cf5e9ca3586a74c194d46acd249a5```

## Version 1.0.1: Switching Protocols (Beta 2)

_OCT 19, 2018_

> AvoidingObstacle2

Add lab(s) in console

Sources:

[Obstacle.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/173f1aad2f8d4a8811dd245fe44b2309f1b4f5a8/Obstacle.java)

[Execut.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/26d6f3481a5e8b83127c26a89635e5a59a35f880/Execut.java)

[Dog_.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/fa3c3e04b2093c1665a907e20173ef5224d97f87/Dog_.java)

**Sha256**: ```4f9aef5f0859600ffbd2cfe3d0c96f028b5b014ef2d72b7e2289c32eb9dcdea2```

**MD5**: ```b2831ea89aa7e93c8cbfacd7a2029f22```

**SHA1**: ```cd66cbdf4529f8c60a7c35658856d75c5791d437```

## Version 1.0.2: Processing (Beta 3)

_OCT 21, 2018_

> AvoidingObstacle3

Introduce GUI

Sources: 

[Obstacle.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/e003eca45557c64a1327843974c8508384513e2b/Obstacle.java)

[Execut.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/39784a5a66e3a82246ae29a7fa816fc3294d9323/Execut.java)

[Dog_.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/b6d744ea8382ceac8412277884602aa86c029d2c/Dog_.java)

[Nani.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/33d7953f0534a3d64fdb7e7a5f359bab5b661ebd/Nani.java)

[Frychicken.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/78eb988d8b8e164550870e47c1a792571d3711f5/Frychicken.java)

**Sha256**: ```6e924362f1172d1c57adfe07412dcf811e6844e4f492a3cab8d7e8e5b61a6bb5```

**MD5**: ```c47bf2defee5665f5aefc89d7f8a503e```

**SHA1**: ```49ca3cf5e35ea5c854e569e332984bb12e32d4a9```

## Version 1.0.3: Early Hints (Beta 4)

_OCT 22, 2018_

> AvoidingOBstacle4

Introduce Obstacle(s)

Bugs 2 Obstacles +

Sources:

[Obstacle.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/758bf25d27b232dd60d316c3f680c84ac2bc59db/Obstacle.java)

[Execut.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/844190e9d32b4b71e8abd5f0a7ffead1a52e1c13/Execut.java)

[Dog_.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/60a221e853a37e748b8ec25a13dd568f3304042d/Dog_.java)

[Nani.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/a5db9f6975f7042b8fdb9d7648ff15894f41a318/Nani.java)

[Frychicken.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/00db4681b0f2e80dcd611c5ae10665f4491ba92d/Frychicken.java)

**Sha256**: ```e8c47dc79636a07a371fda2c6b528c536e907b9c7e39d75284c185c210fc9393```

**MD5**: ```c5402c37c1cc2156e1e2b39369110044```

**SHA1**: ```998e00534971debd065f89b90bbd5ca81f35c009```

## Version 2.0.0: OK (Beta 5)

_OCT 24, 2018_

> AvoidingObstacle5

Fix bugs in beta 4

Sources (Only Bugs fixed):

[Nani.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/8ae74dc22e5502d4103cbff0262b22e19b94be84/Nani.java)

[Execut.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/fc7ad50390c8826801a4c0d9b68c15016e3aaa71/Execut.java)

**Sha256**: ```9748d08873c7ed5d16af105dcf09da0dae02ab6422e41be30357743458e6631e```

**MD5**: ```cd7d722a5f1c6872de02faa9f2b4ae6e```

**SHA1**: ```f0f4c3bf420452d01c122453ffcaa9a486478126```

## Version 2.0.1: Created (Beta 6)

_OCT 25, 2018_

> AvoidingObstracle6

Introduce Debug mode

Sources (Only Changes):

[Execut.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/f4945c9bd7c8ee7b81d3b11e21a4acaa9466a2b4/Execut.java)

[Frychicken.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/08db5bfe6b0fb27521e449a4b63f983e7aee0443/Frychicken.java)

[Nani.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/d101d37b8e8a174bc038f61088194454294c300b/Nani.java)

**Sha256**: ```0e1526da63a39667393b48f4ed6ccd15b8535b7f2dab7b3471303489efc4312c```

**MD5**: ```2691c7419f3804a05771596fc3baa0fd```

**SHA1**: ```d94212eecde29c258fe6ca4a432bacac69deb8db```

## Version 2.0.2: Accepted (Beta 7)

_OCT 26, 2018_

> AvoidingObstacle7

Introduce Loading screen

Several bugs fixed

Sources (Only changes):

[Obstacle.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/5a742ebf627520d7f09ac3648df0206c8986c079/Obstacle.java)

[Execut.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/853adb9e7cd27661bc0687cf63d1fc016bfb7d69/Execut.java)

[Frychicken.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/98582b217aabbce97752844e7ad3606b4504dd52/Frychicken.java)

[Loading.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/891e596116a01aefd20fab10e4e2cd5258fcb516/Loading.java)

[Nani.java](https://bitbucket.org/bobdinh139/art-if-icial-intelligence/raw/1903a4785354b06d102440ca4f32cc687c3c6aae/Nani.java)

**Sha256**: ```bad202a441f8dbfcee7f4be01cfc1bbea8dc834af62a697a46c00b57613f3016```

**MD5**: ```b303bdd80181e31625f2704eddca8018```

**SHA1**: ```d698cebd6805b4bf25740d2bd2941d83677cd2e5```

## Version 2.0.4: No Content

_OCT 27, 2018_

> AO.jar

fixed bugs from beta (can be found in history)

**Sha256**: ```8696301b2db19a1d0a13ba959a2124c3ff9027ae9d313a5c81c0f427d6528e1e```

**MD5**: ```fd2d1af432b24b2a4bc302dd44536976```

**SHA1**: ```ea9bf72b4189b6f1ba81fcf1e10768211c2c9047```

## Version 4.0.4: Not Found

_OCT 27, 2018_

> AOW0.jar

add coloring for object and obstacle(s)

add setup menu (merge debug mode in setup) + redesign setup

clean up code


Execut.java
import javax.swing.*; 
import java.awt.*;
import java.awt.event.KeyAdapter;
import java.awt.event.KeyEvent;
import java.io.IOException;
import java.util.*;
public class Execut  extends Component {
	private static final long serialVersionUID = 1L;
	private JFrame frame;
	private Method_T contentt;
	String msg = "UP";
	private int x = 200;
	private int y = 600;
	Color object = Color.RED;
	Color obstacle = Color.GREEN;
	private boolean debugg = true;
	private boolean sound = true;
	private ArrayList<Integer> xOb = new ArrayList<Integer>();
	private  ArrayList<Integer> yOb = new ArrayList<Integer>();
	int userin = 1;
	int lalala;
	int alala;
	boolean autopilot = false;
	Frychicken fry = new Frychicken();
	public Execut(int userin, Color object, Color obstacle, int i, int b) {
		this.object = object;
		this.obstacle = obstacle;
		lalala = i;
		alala = b;
		this.userin = userin;
	}
	public void assin(boolean debugg, boolean sound, boolean autopilot) throws IOException {
		this.debugg = debugg;
		this.sound = sound;
		this.autopilot = autopilot;
		for (int i =0; i < userin; i++) {
			if (i ==0 || i ==2) {
				xOb.add(RanX());
			}else {
				double rand = Math.random();
				if (rand >= 0.5)
					xOb.add(xOb.get(0) + 237);
				else xOb.add(xOb.get(0)-237);
			}
			if (i==0||i==1)
				yOb.add(RanY());
			else {
				double rand = Math.random();
				if (rand >= 0.5)
					yOb.add(yOb.get(0) + 237);
				else yOb.add(yOb.get(0) - 237);
			}
		}
		Collections.sort(yOb);
		Collections.reverse(yOb);
		if (debugg) {
			for (int i= 0; i< userin; i++){
				System.out.println("x" + (i+1) + " " +xOb.get(i));
				fry.writeLog("x" + (i+1) + " " +xOb.get(i));
				System.out.println("y" + (i+1) +" " +yOb.get(i));
				fry.writeLog("y" + (i+1) +" " +yOb.get(i));

			}
		}
	}

	public int RanX() {
		Random randomness = new Random();
		int rand = randomness.nextInt(300)+30;
		return rand;
	}
	public int RanY() {
		Random randomness = new Random();
		int rand = randomness.nextInt(400)+30;
		return rand;
	}
	int owenn;
	int owenn2;
	class Method_T extends Component {
		private static final long serialVersionUID = 1L;
		int ccc = 0;
		public void paint(Graphics g){

			for (int i =0; i < userin; i++) {
				g.setColor(obstacle);
				g.fillRect(xOb.get(i),yOb.get(i),150,150);
				if ((y >= yOb.get(i) && y<= yOb.get(i) +150) && (x+123 >= xOb.get(i)  && x+123 <= xOb.get(i)+150) && debugg) {
					g.setColor(Color.gray);
					g.fillRect(xOb.get(i),yOb.get(i),150,150);
					owenn = yOb.get(i);
					owenn2 = xOb.get(i);
					if (ccc == 0) {
						System.out.println("Hit at x of object: "+owenn2);
						System.out.println("Hit at y of object: "+owenn);
						fry.writeLog("Hit at x of object: "+owenn2);
						fry.writeLog("Hit at y of object: "+owenn);
						ccc++;
					}
				}
				else if ((y >= yOb.get(i)  && y<= yOb.get(i) +150) && (x >= xOb.get(i)  && x <= xOb.get(i)+150) && debugg) {
					g.setColor(Color.gray);
					g.fillRect(xOb.get(i),yOb.get(i),150,150);
					owenn = yOb.get(i);
					owenn2 = xOb.get(i);
					if (ccc ==0) {
						System.out.println("Hit at x of object: "+owenn2);
						System.out.println("Hit at y of object: "+owenn);
						fry.writeLog("Hit at x of object: "+owenn2);
						fry.writeLog("Hit at y of object: "+owenn);
						ccc++;
					}
				}
			}
			if (debugg) {
				g.setColor(Color.black);
				g.fillOval(x, y, 10, 10);
				g.setColor(Color.black);
				g.fillOval(x+123, y, 10, 10);
				g.setColor(Color.red);
				g.drawString(msg, 300, 750);
				int c = 700;
				int cc = 700;

				for (int i =0; i<userin; i++) {
					g.drawString("Hit at y of object: "+owenn, 250, 660);	
					g.drawString("Hit at x of object :"+owenn2, 250, 675);
					g.drawString("y object: "+y, 150, 660);	
					g.drawString("x object: "+x, 150, 675);
					g.drawString("x"+(i+1)+": "+xOb.get(i), 500, c+=20);
					g.drawString("y"+(i+1)+": "+yOb.get(i), 430, cc+=20);
					g.drawString("lab(s) ran: "+lalala, 60, 720);
					g.drawString("Obstacle(s): "+userin, 60, 700);
					g.drawString("lab(s): "+alala, 60, 680);	
					g.drawString("auto-pilot: "+autopilot, 250, 690);	

				}
			}
			g.setColor(object);
			g.fillRect(x + 40, y +40, 50, 50);
		}
	}
	public void run() {
		frame = new JFrame("AO Simulator" + " @Debug mode: "+debugg + ", sound: "+ sound); 
		frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE); 
		contentt = new Method_T();
		frame.getContentPane().add(BorderLayout.CENTER, contentt);  
		frame.setIconImage(Toolkit.getDefaultToolkit().getImage(getClass().getResource("lollol.png")));
		frame.setResizable(false);
		frame.setSize(600, 800); 
		if (autopilot) {
			frame.addKeyListener(new KeyAdapter() {
				public void keyPressed(KeyEvent event) {
					moveTheTHing(event);
				}
			});
		}
		frame.setFocusable(true); 
		frame.setLocationRelativeTo(null);
		frame.setVisible(true); 
		runpls();
	}


	private boolean right = false;
	private boolean left = false;
	private boolean up = true;
	private boolean down = false;
	private boolean check = true;

	public void moveTheTHing(KeyEvent event) { 

		int key = event.getKeyCode(); 
		int key2 = event.getKeyChar();
		if (key == KeyEvent.VK_LEFT) {
			x-=10;
			check = false;
			msg = "LEFT";
		}
		else if (key == KeyEvent.VK_RIGHT) {
			x+=10;
			check = false;
			msg = "RIGHT";
		}
		else if (key == KeyEvent.VK_UP) {
			y-=20;
			check = false;
			msg = "UP";
		}
		else if (key == KeyEvent.VK_DOWN) {
			y+=20;
			check = false;
			msg = "DOWN";
		}
		if (key2 == 's') {
			up = false;
			msg = "Stop";
		}
		if (key2 == 'c') {
			up = true;
			msg = "Moving";
		}

	}


	public void runpls(){
		while(true) {

			for (int i =0; i < userin; i ++) {
				if ((y >= yOb.get(i) && y<= yOb.get(i) +150) && (x+123 >= xOb.get(i)  && x+123 <= xOb.get(i)+150) && check) {

					left = true;
					right = false;
				}
				else if ((y >= yOb.get(i)  && y<= yOb.get(i) +150) && (x >= xOb.get(i)  && x <= xOb.get(i)+150) && check) {
					left = false;
					right = true;
				}
				if (i==3){
					if (yOb.get(userin-2)-yOb.get(userin-1)-150<= 35 &&( (xOb.get(userin-2) < xOb.get(userin-1) && xOb.get(userin-3) > xOb.get(userin-1))||(xOb.get(userin-3) < xOb.get(userin-1) && xOb.get(userin -2) > xOb.get(userin-1)))){
						down= true;
					}
				}
				int fff = 0;
				if(down){
					y+=3;
					fff++;
					if (fff==50){
						right =!right;
						left =!left;
						fff=0;
						up=true;
					}
				}


				if (left) {
					msg = "LEFT";
					x-=3;
					if (y <= yOb.get(i) +150 && (x+123) < xOb.get(i)  ) { up = true; }
					else up = false;
					if (sound) {
						Soundfx soundfx = new Soundfx();
						soundfx.sound();
					}
				}
				if(right) {
					msg = "RIGHT";
					x+=3;
					if (y <= yOb.get(i) +150 && x > xOb.get(i)  ) { up = true; }
					else up = false;
					if (sound) {
						Soundfx soundfx = new Soundfx();
						soundfx.sound();
					}
				}
				if(up) {
					y--;
					left = false;
					right = false;
				}
				if (y+40+50 <= 0) {
					if (sound) {
						Soundfx soundfx = new Soundfx();
						soundfx.sound();
					}
					frame.setVisible(false); 
					return;
				}
				try {
					Thread.sleep(3); 
				}
				catch (Exception nothing) {
					nothing.printStackTrace();
				} 
				frame.repaint(); 
				right = false;
				left = false;
			}
		}
	}
}

